starlite/middleware/exceptions/debug_response.py

```diff
@@ -0,0 +1,121 @@
+    data = {
+        "line_no": line_no - frame_index + idx,
+        "line": escape(line).replace(" ", "&nbsp"),
+        "line_class": "current_line" if idx == frame_index else "",
+    }
+    return template.format(**data)
+
+
+def create_frame_html(frame: inspect.FrameInfo, collapsed: bool) -> str:
+    """Return html representation of the given frame object including filename
+    containing source code and name of the function being executed."""
+    frame_tpl = (tpl_dir / "frame.html").read_text()
+
+    code_lines: List[str] = []
+    for idx, line in enumerate(frame.code_context or []):
+        code_lines.append(create_line_html(line, frame.lineno, frame.index or 0, idx))
+
+    data = {
+        "file": escape(frame.filename),
+        "line": frame.lineno,
+        "symbol_name": escape(get_symbol_name(frame)),
+        "code": "".join(code_lines),
+        "frame_class": "collapsed" if collapsed else "",
+    }
+    return frame_tpl.format(**data)
+
+
+def create_exception_html(exc: BaseException, frame_limit: int) -> str:
+    """Return html representation of exception frames."""
+    frames = inspect.getinnerframes(exc.__traceback__, frame_limit) if exc.__traceback__ else []
+    result = []
+    for idx, frame in enumerate(reversed(frames)):
+        result.append(create_frame_html(frame=frame, collapsed=(idx > 0)))
+
+    return "".join(result)
+
+
+def create_html_response_contet(exc: Exception, request: "Request", frame_limit: int = 15) -> str:
+    """Return exception traceback in HTML."""
+    exception_data: List[str] = [create_exception_html(exc, frame_limit)]
+    cause = exc.__cause__
+    while cause:
+        cause_data = create_exception_html(cause, frame_limit)
+        cause_header = '<h4 class="cause-header">The above exception was caused by</h4>'
+        cause_error_description = f"<h3><span>{escape(str(cause))}</span></h3>"
+        cause_error = f"<h4><span>{escape(cause.__class__.__name__)}</span></h4>"
+        exception_data.append(
+            f'<div class="cause-wrapper">{cause_header}{cause_error}{cause_error_description}{cause_data}</div>'
+        )
+        cause = cause.__cause__
+
+    scripts = (tpl_dir / "scripts.js").read_text()
+    styles = (tpl_dir / "styles.css").read_text()
+    body_tpl = (tpl_dir / "body.html").read_text()
+    return body_tpl.format(
+        scripts=scripts,
+        styles=styles,
+        error=f"<span>{escape(exc.__class__.__name__)}</span> on {request.method} {escape(request.url.path)}",
+        error_description=escape(str(exc)),
+        exception_data="".join(exception_data),
+    )
+
+
+def create_plain_text_response_contet(exc: Exception) -> str:
+    """Return exception traceback in plain text."""
+    return "".join(format_exception(type(exc), value=exc, tb=exc.__traceback__))
+
+
+def create_debug_response(request: "Request", exc: Exception) -> Response:
```

Comment:
Are we happy having this function as public api, and adding to the reference docs?

I have some exception translation handlers that look like this:

```python
def service_exception_to_http_response(
    request: Request[Any, Any], exc: ServiceException
) -> Response[ExceptionResponseContent]:
    http_exc: type[HTTPException]
    if isinstance(exc, UnauthorizedException):
        http_exc = ForbiddenException
    else:
        http_exc = InternalServerException
    if http_exc is InternalServerException and request.app.debug:
        return _create_error_response_from_starlette_middleware(request, exc)
    return create_exception_response(http_exc())
```
And ideally I'd swap out `_create_error_response_from_starlette_middleware()` for this.

Reply:
@peterschutt can we wait a bit (probably a week) with this? I am working on a separate project that provides more elaborate exception renderer. Most likely function interface will remain the same but I prefer to be 100% sure before declaring something a public API.

Reply:
Yes, of course.

It would be nice if there was a way to push the logic of serving the debug response down into `create_exception_response()` so that users can get that for free when creating their own exception handlers such as above. Issue is that `create_exception_response()` doesn't have a way to check `app.debug`.